[2023-08-30T15:29:20.586+0200] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: de_test_workflow.sparkScript manual__2023-08-30T13:29:18.059576+00:00 [queued]>
[2023-08-30T15:29:20.595+0200] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: de_test_workflow.sparkScript manual__2023-08-30T13:29:18.059576+00:00 [queued]>
[2023-08-30T15:29:20.595+0200] {taskinstance.py:1361} INFO - Starting attempt 1 of 2
[2023-08-30T15:29:20.612+0200] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): sparkScript> on 2023-08-30 13:29:18.059576+00:00
[2023-08-30T15:29:20.616+0200] {standard_task_runner.py:57} INFO - Started process 7190 to run task
[2023-08-30T15:29:20.621+0200] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'de_test_workflow', 'sparkScript', 'manual__2023-08-30T13:29:18.059576+00:00', '--job-id', '3', '--raw', '--subdir', '/Users/france.cama/code/glovo_de_test/dags/workflows.py', '--cfg-path', '/var/folders/bq/3fzm19j96pq0b42q_ghqb6j40000gn/T/tmpbf10a3s9']
[2023-08-30T15:29:20.624+0200] {standard_task_runner.py:85} INFO - Job 3: Subtask sparkScript
[2023-08-30T15:29:20.690+0200] {task_command.py:415} INFO - Running <TaskInstance: de_test_workflow.sparkScript manual__2023-08-30T13:29:18.059576+00:00 [running]> on host mba-francesco.local
[2023-08-30T15:29:20.785+0200] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='francesco' AIRFLOW_CTX_DAG_ID='de_test_workflow' AIRFLOW_CTX_TASK_ID='sparkScript' AIRFLOW_CTX_EXECUTION_DATE='2023-08-30T13:29:18.059576+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-08-30T13:29:18.059576+00:00'
[2023-08-30T15:29:20.799+0200] {spark_submit.py:218} INFO - Could not load connection string spark_standalone_conn, defaulting to yarn
[2023-08-30T15:29:20.800+0200] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master yarn --executor-memory 3g --name arrow-spark ./app/main.py
[2023-08-30T15:29:22.030+0200] {spark_submit.py:491} INFO - 23/08/30 15:29:22 WARN Utils: Your hostname, MBA-Francesco.local resolves to a loopback address: 127.0.0.1; using 192.168.1.62 instead (on interface en0)
[2023-08-30T15:29:22.031+0200] {spark_submit.py:491} INFO - 23/08/30 15:29:22 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2023-08-30T15:29:22.082+0200] {spark_submit.py:491} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2023-08-30T15:29:22.082+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:650)
[2023-08-30T15:29:22.083+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:281)
[2023-08-30T15:29:22.083+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:237)
[2023-08-30T15:29:22.083+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:122)
[2023-08-30T15:29:22.083+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1094)
[2023-08-30T15:29:22.083+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1094)
[2023-08-30T15:29:22.083+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
[2023-08-30T15:29:22.084+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1111)
[2023-08-30T15:29:22.084+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1120)
[2023-08-30T15:29:22.084+0200] {spark_submit.py:491} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2023-08-30T15:29:22.112+0200] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/france.cama/code/glovo_de_test/venv/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/Users/france.cama/code/glovo_de_test/venv/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --executor-memory 3g --name arrow-spark ./app/main.py. Error code is: 1.
[2023-08-30T15:29:22.115+0200] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=de_test_workflow, task_id=sparkScript, execution_date=20230830T132918, start_date=20230830T132920, end_date=20230830T132922
[2023-08-30T15:29:22.122+0200] {standard_task_runner.py:104} ERROR - Failed to execute job 3 for task sparkScript (Cannot execute: spark-submit --master yarn --executor-memory 3g --name arrow-spark ./app/main.py. Error code is: 1.; 7190)
[2023-08-30T15:29:22.151+0200] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-08-30T15:29:22.170+0200] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
